### PR TITLE
Fix cmake config detection

### PR DESCRIPTION
**Pull request by xantares**
Fixes the cmake detection from external projects when dealing with the LIB_SUFFIX var and also cross-compiling aspects, see each commit log for more details.

### CODE REVIEW

**Comment by xantares** [comment]
hi @alalek, do you need some more info about this one ?

File: cmake/OpenCVGenConfig.cmake [hunk 1]
```diff
@@ -128,8 +128,11 @@ endif()
 #  Part 3/3: ${BIN_DIR}/win-install/OpenCVConfig.cmake  -> For use within binary installers/packages
 # --------------------------------------------------------------------------------------------
 if(WIN32)
-  set(OpenCV_INCLUDE_DIRS_CONFIGCMAKE "\"\${OpenCV_CONFIG_PATH}/include\" \"\${OpenCV_CONFIG_PATH}/include/opencv\"")
-  set(OpenCV2_INCLUDE_DIRS_CONFIGCMAKE "\"\"")
+
+  if (CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
+    set(OpenCV_INCLUDE_DIRS_CONFIGCMAKE "\"\${OpenCV_CONFIG_PATH}/include\" \"\${OpenCV_CONFIG_PATH}/include/opencv\"")
+    set(OpenCV2_INCLUDE_DIRS_CONFIGCMAKE "\"\"")
+  endif ()
```
Comment (alalek) [inline cmake/OpenCVGenConfig.cmake]: What are build cases require this condition?
Host OS - ? (Windows / MinGW / etc)
Target OS - ? (Android ?)

BTW, Values are got from [line 101](https://github.com/Itseez/opencv/pull/6269/files#diff-95a1135dd9166d702efe3795b131bb40L101)

Comment (xantares) [inline cmake/OpenCVGenConfig.cmake]: @alalek, the target os is alway win32,
I differentiate the case when the host os is linux (cross-compiling), in which case the values are taken from previous values as you mention, and the case when the host is windows (we go inside that if).

Comment (alalek) [inline cmake/OpenCVGenConfig.cmake]: Thanks!
There is #5980 which should handle this case (and case below) without a magic of "if"s.

About a fix above with LIB_SUFFIX (same as in #5940): it changes current behavior, so I need to ask our teams who uses the current "share/OpenCV" way.

Comment (xantares) [inline cmake/OpenCVGenConfig.cmake]: @alalek 
Wow that's quite some changes, maybe I should test it after you merge it.
About LIB_SUFFIX, that shouldn't change anything for client libs, only packaging.

**Comment by alalek** [comment]
@xantares Could you try #5980 and specify `LIB_SUFFIX` via CMake command line option ("-D" parameter)?

**Comment by xantares** [comment]
@alalek I've got an error: seems opencv wants contrib now, how can I disable this ?
```
CMake Error at cmake/OpenCVModule.cmake:295 (message):
  No extra modules found in folder:
  /home/xantares/projects/aur-scripts/opencv/src/opencv_contrib-3.1.0/modules
```

**Comment by alalek** [comment]
Try to build from scratch (or in new build directory).
Or run CMake and remove `OPENCV_EXTRA_MODULES_PATH` parameter:
```
cmake -UOPENCV_EXTRA_MODULES_PATH .
```

**Comment by vpisarev** [comment]
this PR will be recreated because of CI issues

**Pull request closed without merging**